pkg/api/v1/types.go

```diff
@@ -999,6 +1016,10 @@ type ServicePort struct {
 	// target Pod's container ports.  If this is not specified, the value
 	// of Port is used (an identity map).
 	TargetPort util.IntOrString `json:"targetPort" description:"the port to access on the pods targeted by the service; defaults to the service port"`
+
+	// Optional: A port that we will listen on, on every minion.
+	// This allows services to be exposed without necessarily using a load balancer.
+	PublicPort int `json:"publicPort" description:"the port number on which this service is exposed outside the cluster"`
```

Comment:
I agree with C, we have this currently (note that we could get visibility type != cluster if CreateExternalLoadBalancer=true):

```
	if service.Spec.Visibility == api.VisibilityTypeCluster {
		for i := range service.Spec.Ports {
			if service.Spec.Ports[i].PublicPort != 0 {
				allErrs = append(allErrs, errs.NewFieldInvalid("spec.ports", service.Spec.Ports[i], "cannot specify a public port with cluster-visibility services"))
			}
		}
	}
```